src/coreclr/jit/gentree.cpp

```diff
@@ -21967,3 +21967,23 @@ bool GenTree::IsInvariant() const
     GenTree* lclVarTree = nullptr;
     return OperIsConst() || Compiler::impIsAddressInLocal(this, &lclVarTree);
 }
+
+//------------------------------------------------------------------------
+// IsNeverNegative: returns true if the given tree is known to be never
+//      negative such as: casts to unsiged, UDIV, UMOD, Array.Length
+//
+// Arguments:
+//    comp - Compiler object, needed for IntegralRange::ForNode
+//
+// Return Value:
+//      true if the given tree is known to be never negative
+//
+bool GenTree::IsNeverNegative(Compiler* comp) const
+{
+    if (IsCnsIntOrI())
+    {
+        return AsIntCon()->IconValue() >= 0;
+    }
```

Comment:
Could be `IsIntegralConst` and `IntegralValue() >= 0` to handle `CNS_LNG` as well.

Reply:
ah, it was like that in the previous commit, let me revert it :)